### DISCUSSION TITLE
Cannot start the development environment on mac M1

### DISCUSSION THREAD

**Discussion by abhavgoel** [coollabsio/coolify#2884] [Improvement Requests]
Hi, I am trying to clone and build the repo in my local machine, but when i do `sudo spin up` I facing the following error - 

[spin] ⚡️ Skipping docker image pull because it ran earlier.
WARN[0000] The "PUSHER_HOST" variable is not set. Defaulting to a blank string.
WARN[0000] The "PUSHER_PORT" variable is not set. Defaulting to a blank string.
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
2024/07/19 20:50:10 must use ASL logging (which requires CGO) if running as root
[+] Running 0/0
 ⠋ vite Pulling                                                            0.0s
 ⠋ postgres Pulling                                                        0.0s
 ⠋ testing-host Pulling                                                    0.0s
 ⠋ redis Pulling                                                           0.0s
 ⠋ mailpit Pulling                                                         0.0s
 ⠋ minio Pulling                                                           0.0s
error getting credentials - err: exit status 1, out: ``

I am using an M1 mac. If someone else has faced this issue kindly provide a solution for the same.

**Comment by peaklabs-dev** [comment]
@abhavgoel Do you have a .env file in the repository (duplicate this file https://github.com/coollabsio/coolify/blob/main/.env.development.example) , otherwise it will not work?

**Comment by abhavgoel** [comment]
Yes I have everything, but it wont just run on M1 serier or maybe higher.
Could you please draft a documentation for running dev environment on mac?

Abhav goel

**Comment by peaklabs-dev** [comment]
@abhavgoel I have completely rewritten the CONTRIBUTING.md file in this PR: https://github.com/coollabsio/coolify/pull/3015 this will be added to Coolify in the next version.

**Reply by andrasbacsai** [reply]
The modifications will be released in #3058

**Comment by abhavgoel** [comment]
Thanks for the update.

**Reply by peaklabs-dev** [reply]
My rewrite of the contributing guide is now live. Please follow these steps and let me know if it works and if it is understandable: https://github.com/coollabsio/coolify/blob/main/CONTRIBUTING.md
